domain/application/service/service.go

```diff
@@ -111,6 +113,12 @@ type SupportedFeatureProvider interface {
 	environs.SupportedFeatureEnumerator
 }
 
+// KubernetesBroker contains methods from the caas.Broker interface
+// used by the application provider service.
+type KubernetesBroker interface {
```

Comment:
A KubernetesBroker is a very different thing to what you are asking for. The provider tracker is what deals with brokers and their construction. You are after an implementation. Would this not better called CAASApplicationProvider or something ?